app/controllers/api/v1/api_keys_controller.rb

```diff
@@ -48,6 +48,8 @@ def update
 
   def check_mfa(user)
     if user&.mfa_gem_signin_authorized?(otp)
+      return render_error_if_user_mfa_requirement_unmet(user) if user&.mfa_required_not_yet_enabled? || user&.mfa_required_weak_level_enabled?
```

Comment:
We had to add those because we would return no matter what otherwise, which would mean the yield would be unaccessible. We have to return because otherwise we might get multiple renders...that's what we think we have to do anyway.

Reply:
Oh I see, that makes sense. Though it isn't great to have this check and go through the same checks in `render_error_if_user_mfa_requirement_unmet(user)` (a possible sign that `render_error...` is a bit too complicated). 

I believe this could be solved with the suggestion given [here](https://github.com/rubygems/rubygems.org/pull/3155#discussion_r929093198).

```ruby
# method names up for refinement
return render_mfa_setup_required if user.mfa_required_not_yet_enabled?
return render_mfa_strong_level_required if user.mfa_required_weak_level_enabled?
```